tests/unit/customizations/s3/__init__.py

```diff
@@ -11,130 +11,90 @@
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
 import os
-import random
-from awscli.testutils import unittest
-import string
 
-from awscli.compat import six
+import mock
 from mock import patch, Mock
 
-
-class S3HandlerBaseTest(unittest.TestCase):
-    pass
-
-
-def make_loc_files():
+from awscli.compat import six
+from awscli.testutils import BaseAWSCommandParamsTest, FileCreator
+
+
+class S3HandlerBaseTest(BaseAWSCommandParamsTest):
+    def setUp(self):
+        super(S3HandlerBaseTest, self).setUp()
+        self.session = self.driver.session
+        self.driver.session.register('before-call', self.before_call)
+        self.driver.session.register('before-parameter-build',
+                                     self.before_parameter_build)
+        self.client = self.session.create_client('s3', 'us-east-1')
+        self.source_client = self.session.create_client('s3', 'us-east-1')
+        self.file_creator = FileCreator()
+
+    def tearDown(self):
+        super(S3HandlerBaseTest, self).tearDown()
+        clean_loc_files(self.file_creator)
+
+    def run_s3_handler(self, s3_handler, tasks):
+        self.patch_make_request()
+        captured_stderr = six.StringIO()
+        captured_stdout = six.StringIO()
+        with mock.patch('sys.stderr', captured_stderr):
```

Comment:
This looks identical to the code in testutils.py.  What about extracting the captured stdout/stderr logic into a separate function in that module so we can share code?  I'd think something like would work:

```
with testutils.capture_output() as captured:
    <your code to run>
print(captured.stdout.getvalue())
print(captured.stderr.getvalue())
```